From c1cc7bba8b798680607567132497e7f29184c5ce Mon Sep 17 00:00:00 2001
From: Daniel Leung <daniel.leung@intel.com>
Date: Wed, 2 Oct 2013 11:31:05 +0000
Subject: wifi: Add ability to use custom wifi code

Adds a switch so that a different source file for wifi can be
used instead of the default one from AOSP. This gives us the
flexibility of having different wifi code for various build
targets.

One may argue that the code can be built as a static library
and be included here. However, since nothing in here refers
to functions in the wifi code, the linker simply does not
include the library. So the source has to be built at the
same time as other source files here.

Also by moving the code away from this project, this makes
it easier for us to move forward to upstream commits.

Category: Device Enablement
Upstream Candidate: No
Domain: AOSP-wifi
Issue: AXIA-1242, ABRO-114, ABRO-151

Change-Id: I222f4bf14683cdc3a72857b740340b10d2325ebc
---
 wifi/Android.mk |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/wifi/Android.mk b/wifi/Android.mk
index cde99d7..daed62d 100644
--- a/wifi/Android.mk
+++ b/wifi/Android.mk
@@ -1,5 +1,9 @@
 # Copyright 2006 The Android Open Source Project
 
+ifneq ($(BOARD_CUSTOM_WIFI_HAL_SRC),)
+LOCAL_SRC_FILES += ../../$(BOARD_CUSTOM_WIFI_HAL_SRC)
+else
+
 ifdef WIFI_DRIVER_MODULE_PATH
 LOCAL_CFLAGS += -DWIFI_DRIVER_MODULE_PATH=\"$(WIFI_DRIVER_MODULE_PATH)\"
 endif
@@ -27,4 +31,6 @@ endif
 
 LOCAL_SRC_FILES += wifi/wifi.c
 
+endif
+
 LOCAL_SHARED_LIBRARIES += libnetutils
-- 
1.7.4.1

